command/git.go

```diff
@@ -267,8 +279,8 @@ func (c *gitFetchProject) buildModuleCloneCommand(conf *model.TaskConfig, opts c
 		gitCommands = append(gitCommands,
 			fmt.Sprintf(`git fetch origin "pull/%s/merge:%s"`, modulePatch.PatchSet.Patch, branchName),
 			fmt.Sprintf("git checkout '%s'", branchName),
-			fmt.Sprintf("git reset --hard %s", modulePatch.Githash),
 		)
+		gitCommands = append(gitCommands, c.gitResetWithUnshallow(modulePatch.Githash)...)
 	} else {
 		gitCommands = append(gitCommands, fmt.Sprintf("git checkout '%s'", ref))
```

Comment:
I can't figure out why we would do this instead of just calling `git reset`. Does anyone know a reason why we can't get rid of this branch? @ybrill Is this necessary for the commit queue?

Reply:
I think it is possible replace the checkout with reset --hard

Reply:
factoring both the resets out of the if/else would take refactoring the Execute function to pass the expected commit instead of the branch name in the case of a PR commit queue patch.

Reply:
I think checkout leaves you in a detached head state which maybe you want?

Reply:
@ybrill You can reset to a branch though. Does the commit queue check out a different branch? I think this will break with `git clone --depth <n> --branch <branch>` if we need to check out a different branch from the one passed to the clone, since a shallow clone does not get every branch, and `--unshallow` doesn't either.

Reply:
discussed in person